src/tasks/withdraw/value.ts

```diff
@@ -0,0 +1,115 @@
+import axios from "axios";
+import { BigNumber, constants } from "ethers";
+
+import { OrderKind } from "../../ts";
+import { TokenDetails } from "../ts/erc20";
+
+interface ApiTradeQuery {
+  network: string;
+  baseToken: string;
+  quoteToken: string;
+  kind: OrderKind;
+  amount: BigNumber;
+}
+
+interface PricedToken extends TokenDetails {
+  // Overrides existing field in TokenDetails. The number of decimals must be
+  // known to estimate the price
+  decimals: number;
```

Comment:
Isn't this redundant if it's already on TokenDetails?

Reply:
`TokenDetails` contains `decimals: number | undefined` because in principle ERC20 tokens need not provide a `decimal()` interface, but I need the decimals to compute the price, so here I'm restricting the type to always have decimals.